File: carrera-sdk/producer/java/carrera-producer-sdk/src/main/java/com/xiaojukeji/carrera/producer/tx/TxBusinessMessageBuilder.java
package com.xiaojukeji.carrera.producer.tx;

import com.xiaojukeji.carrera.producer.MessageBuilder;
import com.xiaojukeji.carrera.thrift.Result;


public class TxBusinessMessageBuilder {

    private MessageBuilder messageBuilder;

    public TxBusinessMessageBuilder(MessageBuilder messageBuilder) {
        this.messageBuilder = messageBuilder;
    }

    public Result send() {
        return this.messageBuilder.send();
    }
}

File: carrera-sdk/producer/go/src/carrera/txBusinessMessageBuilder.go
package carrera

import "carrera/CarreraProducer"

type TxBusinessMessageBuilder struct {
	messageBuilder *MessageBuilder
}

func NewTxBusinessMessageBuilder(messageBuilder *MessageBuilder) *TxBusinessMessageBuilder {
	return &TxBusinessMessageBuilder{messageBuilder:messageBuilder}
}

func (txBusinessMessageBuilder *TxBusinessMessageBuilder) Send() *CarreraProducer.Result {
	return txBusinessMessageBuilder.messageBuilder.Send()
}
